src/devices/Mcp25xxx/Mcp25xxx.cs

```diff
@@ -292,11 +293,21 @@ public void SetMode(OperationMode operationMode)
         }
 
         /// <summary>
-        /// Read messages from all buffers
+        /// Read arrived messages
         /// </summary>
-        /// <param name="byteCount">Number of bytes to read. This must be one or more to read.</param>
         /// <returns>List of messages received</returns>
-        public List<byte[]> ReadMessages(int byteCount = TransmitBufferMaxSize)
+        public CanMessage[] ReadMessages()
+        {
+            var buffersData = ReadAllBuffers(TransmitBufferMaxSize);
+            return buffersData.Select(CanMessage.CreateFromBytes).ToArray();
```

Comment:
The only problem I see here is that this will now throw an `ArgumentException` when (for whatever reason) the input data is garbage. This is confusing and should probably be wrapped into some other error.